plugins/destination/postgresql/client/migrate.go

```diff
@@ -326,50 +346,51 @@ func (c *Client) addColumn(ctx context.Context, tableName string, column schema.
 	return nil
 }
 
-func (c *Client) createTableIfNotExist(ctx context.Context, table *schema.Table) error {
+func (c *Client) createTableIfNotExist(ctx context.Context, table *arrow.Schema) error {
 	var sb strings.Builder
-	tableName := pgx.Identifier{table.Name}.Sanitize()
+	tName := schema.TableName(table)
+	tableName := pgx.Identifier{tName}.Sanitize()
 	sb.WriteString("CREATE TABLE IF NOT EXISTS ")
 	sb.WriteString(tableName)
 	sb.WriteString(" (")
-	totalColumns := len(table.Columns)
+	totalColumns := len(table.Fields())
 
 	primaryKeys := []string{}
-	for i, col := range table.Columns {
+	for i, col := range table.Fields() {
 		pgType := c.SchemaTypeToPg(col.Type)
 		if pgType == "" {
```

Comment:
Unrelated to this PR, but can `SchemaTypeToPg` return an empty string? Looks like we default to `text`